### PR TITLE
Add validation of backend memory support when executing with memory=True

### PR DESCRIPTION
**Pull request by jwoehr**
### Summary
Test backend for `memory=True` configuration property if that kwarg is passed to `execute()`

### Details and comments
Throw QiskitException if `execute(memory=True)` and `backend.configuration().memory is False`

### CODE REVIEW

**Comment by jwoehr** [comment]
Addresses #3251

**Comment by jwoehr** [comment]
> This looks good. Can you move the check into `qiskit.compiler.assemble._parse_common_args`?

Done.

File: qiskit/compiler/assemble.py [hunk 1]
```diff
@@ -178,11 +178,19 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
...
+        memory.
     """
     # grab relevant info from backend if it exists
     backend_config = None
     if backend:
         backend_config = backend.configuration()
+        # check for memory flag applied to backend that does not support memory
+        if memory and not backend_config.memory:
+            raise QiskitError("memory not supported by backend {}"
+                              .format(backend.properties().backend_name))
```
Comment (taalexander) [inline qiskit/compiler/assemble.py]: The name is available in the configuration.

Comment (jwoehr) [reply qiskit/compiler/assemble.py]: Done.

**Pull request merged**